js/components/common/react_modal.js.jsx

```diff
@@ -50,9 +50,16 @@ class CourseModal extends React.Component {
   }
 
   closeModal() {
+    this.props.callbackModal();
     this.setState({ modalIsOpen: false });
   }
 
+  componentDidUpdate (prevProps) {
+    if ((prevProps !== this.props) && (this.props.courseChange !== "")) {
+      this.openModal(this.props.courseChange);
```

Comment:
Merge this function with the `openModal` function above. The part to keep is the call to `getCourse`.